### PR TITLE
24423 Added conditional dashboard redirect

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* bcgov/entity#24423

*Description of changes:*
- app version = 7.4.5
- on start, redirect to new dashboard if FF is True

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -448,6 +448,15 @@ export default class App extends Mixins(
...
 
+    // now that LaunchDarkly has been updated (ie, in case of user targeting),
+    // check whether to use this Entity Dashboard or the new Business Dashboard
+    if (GetFeatureFlag('use-business-dashboard')) {
+      const identifier = (this.businessId || this.tempRegNumber)
+      const dashboardUrl = `${this.getBusinessDashUrl}/${identifier}${this.$route.fullPath}`
+      navigate(dashboardUrl)
+      return
+    }
+
```
Comment (severinbeauvais) [inline src/App.vue:459]: I hoped to add this code to main.ts, ie, before Vue was started, but for FF targeting, the LD user first had to be updated (line 443).

This code redirects to the new business dashboard if the FF is True. This means we don't have to update Namerequest UI or Auth Web (My Business Registry) to redirect conditionally.

Note that `$route.fullPath` includes any URL parameters (such as Filing ID and Account).

If you want to test this, contact me with your IDIR and I'll update LaunchDarkly to target you only.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-filings-dev--pr-703-athj4a1p.web.app

File: src/App.vue (hunk 1 above)
Comment (severinbeauvais) [reply src/App.vue:459]: Sample redirect URL when :

image

**Review by hfekete — APPROVED** [review]
Wow, thanks for this quick PR !!

**Review by JazzarKarim — APPROVED** [review]
LGTM 👍 👍

**Comment by ketaki-deodhar** [comment]
got to this a bit late but looks good

**Pull request merged**